src/main/java/sim/coffee/OrderList.java

```diff
@@ -1,5 +1,137 @@
 package sim.coffee;
 
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.LinkedList;
+import java.util.Scanner;
+
 public class OrderList {
+
+	LinkedList<Order> OrderList;	
+	// LinkedList has been declared
+	
+	public OrderList()		// Constructor for orderLisr=t is created
+	{
+		OrderList= new LinkedList<Order>();
+	}	
+	
+	public void readFile(String fileName) {  /* 1. Is process line method same as that in menu class
+											    2. Method to read CSV file*/
+		
+		File inputFileObject = new File("input.txt"); 
+		Scanner scannerObject;
+		
+		try {
+			
+			scannerObject = new Scanner(inputFileObject);
+			for(int i=0;scannerObject.hasNextLine();i++)
+			{
+				processLine(scannerObject.nextLine());
+			}
+			
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
+		} 
+		}
     
+		public void processLine(String Line)
+		{
+			try
+			{
+				if (Line.split(",").length>=2)
+				{
+					String[] words = Line.split(",");
+					
+					if (words[3].matches("^B"))
+					{
+						OrderBeverage orderBeverage = new OrderBeverage(word[3],word[4],word[5],new MenuItem(word[2],
```

Comment:
We can tidy the code up a bit here to be more readable, but looks like you have the right idea.

We actually might have overlooked something when planning this as I'm not sure we'd considered that a MenuItem will need to be instantiated here too 🤔